jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/FormField.java

```diff
@@ -0,0 +1,322 @@
+
+    @JDIAction("Get hint for '{name}' with index {0}")
+    public String hint(int index) {
+        UIElement hint = getFormFields().get(index).find(By.cssSelector("mat-hint"));
+        return hint.getText();
+    }
+
+    @JDIAction("Get placeholder for '{name}' with index {0}")
+    public String placeholder(int index) {
+        String placeholder = "placeholder";
+        String type = getType(index);
+        WebElement element = getFormFields().get(index).find(By.xpath("//" + type));
+        UIElement actualElement = new UIElement(element);
+        return (actualElement.hasAttribute(placeholder)) ? actualElement.getAttribute(placeholder) : "";
+    }
+
+    @JDIAction("Get label for '{name}' with index {0}")
+    public String label(int index) {
+        UIElement label = getFormFields().get(index).find(By.xpath("//mat-label"));
+        return label.getText();
+    }
+
+    @JDIAction("Get error for input in '{name}' with index {0}")
+    public String inputError(int index) {
+        UIElement error = getInputs().get(index).
+            find(By.xpath("//ancestor::mat-form-field[@" + containerAttribute + "]//mat-error"));
+        return error.getText();
+    }
+
+    @JDIAction("Get error for '{name}' with index {0}")
+    public String error(int index) {
+        UIElement error = getFormFields().get(index).find(By.xpath("//mat-error"));
+        return error.getText();
+    }
+
+    @JDIAction("Get focus out from '{name}'")
+    public void focusOut() {
+        getContainer().click();
+    }
+
+    @JDIAction("Clear value from input in '{name}' with index {0}")
+    public void clearInput(int index) {
+        getInputs().get(index).sendKeys(Keys.CONTROL + "a");
+        getInputs().get(index).sendKeys(Keys.DELETE);
+        focusOut();
+    }
+
+    @JDIAction("Clear value from text area in '{name}' with index {0}")
+    public void clearTextArea(int index) {
+        getTextAreas().get(index).sendKeys(Keys.CONTROL + "a");
+        getTextAreas().get(index).sendKeys(Keys.DELETE);
+        focusOut();
+    }
+
+    public String getType(int index) {
+        String type = "";
+        getContainer();
+        String currentXpath = "(" + matFormFieldPrefix + containerAttribute + "])[" + index + "]";
+        try {
+            getDriver().findElement(By.xpath(currentXpath + "//input"));
+            type = "input";
+        } catch (Exception e1) {
+            try {
+                getDriver().findElement(By.xpath(currentXpath + "//mat-select"));
+                type = "mat-select";
+            } catch (Exception e2) {
+                try{
+                    getDriver().findElement(By.xpath(currentXpath + "//textarea"));
+                    type = "textarea";
+                } catch (Exception e3) {
+                    JDILogger logger = new JDILogger();
+                    logger.error(e3.toString());
+                }
+            }
+        }
+        return type;
+    }
+
+    public WebList getFormFields() {
```

Comment:
This method is only used in the current class. Maybe the modifier should be protected.

Reply:
Done.